src/packages/@ncigdc/utils/download/index.js

```diff
@@ -251,37 +251,26 @@ const download = ({ url, params, method = 'GET', altMessage = false }) => {
     '',
   );
 
-  const formHtml = `<form
-      method="${method.toUpperCase()}"
-      id="${formId}"
-      action="${url}"
-      style="display: none"
-    >
-      ${fields}
-    </form>`;
-
-  $(
-    `<iframe
-      id="${iFrameId}"
-      style="display: none"
-      src="about:blank"
-      onload="this.__frame__loaded = true;">
-    </iframe>`,
-  )
-    // Appending to document body to allow navigation away from the current
-    // page and downloads in the background
-    .appendTo('body');
-
-  const iFrame = $(`#${iFrameId}`);
-  iFrame[0].__frame__loaded = false;
-
-  iFrame.ready(() => {
-    const iFrameBody = iFrame.contents().find('body');
-    iFrameBody.append(formHtml);
-
-    const form = iFrameBody.find(`#${formId}`);
-    form.submit();
-  });
+  const iFrame = document.createElement('iframe');
+
+  iFrame.style.display = 'none';
+  iFrame.src = 'about:blank';
+  iFrame.onload = function() {
+    this.__frame__loaded = true;
+  };
+  // Appending to document body to allow navigation away from the current
+  // page and downloads in the background
+  document.body.appendChild(iFrame);
+  iFrame.__frame__loaded = false;
+
+  const form = document.createElement('form');
+  form.method = method.toUpperCase();
+  form.action = url;
+  form.innerHTML = fields;
+
+  getBody(iFrame).appendChild(form);
+
+  form.submit();
```

Comment:
Theres no funny business(loading async stuff) going on inside the form so waiting shouldn't be required.
Definitely give it a test though, everything worked for me.